libs/renterd-types/src/types.ts

```diff
@@ -156,6 +157,50 @@ export type S3AuthenticationSettings = {
   }
 }
 
+export type Pin = {
+  pinned: boolean
+  value: number
+}
+
+export type GougingSettingsPins = {
+  maxStorage: Pin
+  maxDownload: Pin
+  maxUpload: Pin
+  maxRPCPrice: Pin
+}
+
+export type AutopilotPins = {
+  allowance: Pin
+}
+
+// PricePinSettings holds the configuration for pinning certain settings to
+// a specific currency (e.g., USD). It uses a Forex API to fetch the current
+// exchange rate, allowing users to set prices in USD instead of SC.
+export type PricePinSettings = {
+  // Enabled can be used to either enable or temporarily disable price
+  // pinning. If enabled, both the currency and the Forex endpoint URL
+  // must be valid.
+  enabled: boolean
+
+  // Currency is the external three-letter currency code.
+  currency: CurrencyId | ''
```

Comment:
Empty string will return an error

Reply:
<img width="320" alt="Screenshot 2024-08-08 at 9 28 26 AM" src="https://github.com/user-attachments/assets/a1da043f-ae70-4a2d-a619-1feb46b06ffc">

- The empty values for currency and forex are empty string, that is what that type is supposed to capture.
- Additionally, for some reason even with this change: https://github.com/SiaFoundation/renterd/commit/e0b464f564b0cba1b6c76edc040ed673b26b123c I don't get an object for the autopilots.allowance key, I was expecting one similar to the gouging ones.

Reply:
Hm yeah the pin settings are not aware of existing autopilots since they are just a json blob in the db. I opened a [PR](https://github.com/SiaFoundation/renterd/pull/1424) to populate the map with all registered autopilots in the bus.

Reply:
Cool, I will add an null check to the other PR for now.